datajunction-server/datajunction_server/construction/build.py

```diff
@@ -1161,21 +1181,31 @@ def build_ast(  # pylint: disable=too-many-arguments
 def metrics_to_measures(
     session: Session,
     metric_nodes: List[Node],
-) -> DefaultDict[str, Set[str]]:
+) -> Tuple[DefaultDict[str, Set[str]], DefaultDict[str, Set[str]]]:
     """
-    Returns a mapping between each metric node and a list of necessary measures for that metric.
+    For the given metric nodes, returns a mapping between the metrics' referenced parent nodes
+    and the list of necessary measures to extract from the parent node.
+    The structure is:
+    {
+        "parent_node_name1": ["measure_columnA", "measure_columnB"],
+        "parent_node_name2": ["measure_columnX"],
+    }
     """
     ctx = CompileContext(session, DJException())
     metric_to_measures_mapping = collections.defaultdict(set)
+    parents_to_measures = collections.defaultdict(set)
     for metric_node in metric_nodes:
         metric_ast = parse(metric_node.current.query)
         metric_ast.compile(ctx)
         for col in metric_ast.find_all(ast.Column):
             if col.table:  # pragma: no cover
-                metric_to_measures_mapping[col.table.dj_node.name].add(  # type: ignore
+                parents_to_measures[col.table.dj_node.name].add(  # type: ignore
+                    col.alias_or_name.name,
+                )
+                metric_to_measures_mapping[metric_node.name].add(
                     col.alias_or_name.name,
                 )
-    return metric_to_measures_mapping
+    return parents_to_measures, metric_to_measures_mapping
```

Comment:
I agree we don't need the `_mapping` suffix. You may want to adjust the 2nd variable as well.

Reply:
good point, removed!